### PR TITLE
Renames operator to staking provider in TokenStaking

### PR DESCRIPTION
**Pull request by vzotova**
See #66

### CODE REVIEW

File: contracts/staking/TokenStaking.sol [hunk 1]
```diff
@@ -1590,36 +1693,44 @@ contract TokenStaking is Initializable, IStaking, Checkpoints {
...
         uint256 newLength = length - numberToDelete;
         while (index < newLength && deleted < numberToDelete) {
-            address application = operatorStruct.authorizedApplications[index];
-            if (operatorStruct.authorizations[application].authorized == 0) {
-                operatorStruct.authorizedApplications[index] = operatorStruct
-                    .authorizedApplications[length - deleted - 1];
+            address application = stakingProviderStruct.authorizedApplications[
+                index
+            ];
+            if (
```
Comment (derekpierre) [inline contracts/staking/TokenStaking.sol:1712]: Nice algo.

File: contracts/staking/IApplication.sol [hunk 2]
```diff
@@ -19,37 +19,37 @@ pragma solidity 0.8.9;
...
-///         operator are eligible to slash the stake delegated to that operator.
+///         provider are eligible to slash the stake delegated to that provider.
 interface IApplication {
     /// @notice Used by T staking contract to inform the application that the
-    ///         authorized amount for the given operator increased.
+    ///         authorized amount for the given provider increased.
     ///         The application may do any necessary housekeeping.
     function authorizationIncreased(
-        address operator,
+        address stakingprovider,
```
Comment (derekpierre) [inline contracts/staking/IApplication.sol:28]: Any reason we don't use camel case here (i.e. `stakingProvider`) but we do everywhere else?

File: docs/rfc-1-staking-contract.adoc [hunk 3]
```diff
@@ -277,27 +277,27 @@ out of the scope of this document.
...
-==== `stakeNu(address operator, address payable beneficiary, address authorizer) external`
+==== `stakeNu(address stakingProvider, address payable beneficiary, address authorizer) external`
 
 Copies delegation from the legacy NU staking contract to T staking contract,
 additionally appointing beneficiary and authorizer roles. Caches the amount
 staked in NU staking contract. Can be called only by the original delegation
 owner.
 
-==== `refreshKeepStakeOwner(address operator) external onlyOwnerOf(operator)`
+==== `refreshKeepStakeOwner(address stakingProvider) external onlyOwnerOf(stakingprovider)`
```
Comment (derekpierre) [inline docs/rfc-1-staking-contract.adoc:300]: mismatched camel case for `stakingProvider` vs `stakingprovider`? If so, other places lower down as well.

File: contracts/staking/IApplication.sol (hunk 2 above)
Comment (vzotova) [reply contracts/staking/IApplication.sol]: oh, thank you, I thought I fixed it but apparently missed few places
✔️

File: docs/rfc-1-staking-contract.adoc (hunk 3 above)
Comment (vzotova) [reply docs/rfc-1-staking-contract.adoc]: ✔️

**Review by cygnusv — APPROVED** [review]
You rock @vzotova

**Review by pdyraga — APPROVED** [review]

File: contracts/staking/IStaking.sol [hunk 4]
```diff
@@ -36,12 +36,12 @@ interface IStaking {
     //
 
     /// @notice Creates a delegation with `msg.sender` owner with the given
-    ///         operator, beneficiary, and authorizer. Transfers the given
+    ///         provider, beneficiary, and authorizer. Transfers the given
```
[the remark this answers is not in the document]
Comment (pdyraga) [reply contracts/staking/IStaking.sol:39]: Done in https://github.com/threshold-network/solidity-contracts/pull/71.

File: contracts/staking/IStaking.sol [hunk 5]
```diff
@@ -51,20 +51,20 @@ interface IStaking {
     ///         staking contract. No tokens are transferred. Caches the active
     ///         stake amount from KEEP staking contract. Can be called by
     ///         anyone.
-    function stakeKeep(address operator) external;
+    function stakeKeep(address stakingProvider) external;
```
[the remark this answers is not in the document]
Comment (pdyraga) [reply contracts/staking/IStaking.sol:54]: Done in https://github.com/threshold-network/solidity-contracts/pull/71.

File: contracts/staking/IApplication.sol [hunk 6]
```diff
@@ -19,37 +19,37 @@ pragma solidity 0.8.9;
...
     function authorizationIncreased(
-        address operator,
+        address stakingProvider,
         uint96 fromAmount,
         uint96 toAmount
     ) external;
 
     /// @notice Used by T staking contract to inform the application that the
-    ///         given operator requested to decrease the authorization amount.
+    ///         given provider requested to decrease the authorization amount.
```
[the remark this answers is not in the document]
Comment (pdyraga) [reply contracts/staking/IApplication.sol:34]: Done in https://github.com/threshold-network/solidity-contracts/pull/71.

File: contracts/staking/IStaking.sol [hunk 7]
```diff
@@ -51,20 +51,20 @@ interface IStaking {
...
     function stakeNu(
-        address operator,
+        address stakingProvider,
         address payable beneficiary,
         address authorizer
     ) external;
 
     /// @notice Refresh Keep stake owner. Can be called only by the old owner.
-    function refreshKeepStakeOwner(address operator) external;
+    function refreshKeepStakeOwner(address stakingProvider) external;
```
[the remark this answers is not in the document]
Comment (pdyraga) [reply contracts/staking/IStaking.sol:67]: Done in https://github.com/threshold-network/solidity-contracts/pull/71.

File: contracts/staking/TokenStaking.sol [hunk 8]
```diff
@@ -305,33 +310,37 @@ contract TokenStaking is Initializable, IStaking, Checkpoints {
     ///         staking contract. No tokens are transferred. Caches the active
     ///         stake amount from KEEP staking contract. Can be called by
     ///         anyone.
-    function stakeKeep(address operator) external override {
-        require(operator != address(0), "Parameters must be specified");
-        OperatorInfo storage operatorStruct = operators[operator];
+    function stakeKeep(address stakingProvider) external override {
```
[the remark this answers is not in the document]
Comment (pdyraga) [reply contracts/staking/TokenStaking.sol:313]: Done in https://github.com/threshold-network/solidity-contracts/pull/71.

File: contracts/staking/TokenStaking.sol [hunk 9]
```diff
@@ -341,62 +350,73 @@ contract TokenStaking is Initializable, IStaking, Checkpoints {
...
             tAmount
         );
     }
 
     /// @notice Refresh Keep stake owner. Can be called only by the old owner.
-    function refreshKeepStakeOwner(address operator) external override {
-        OperatorInfo storage operatorStruct = operators[operator];
-        require(operatorStruct.owner == msg.sender, "Caller is not owner");
-        address newOwner = keepStake.resolveOwner(operator);
+    function refreshKeepStakeOwner(address stakingProvider) external override {
```
[the remark this answers is not in the document]
Comment (pdyraga) [reply contracts/staking/TokenStaking.sol:397]: Done in https://github.com/threshold-network/solidity-contracts/pull/71.

File: contracts/staking/TokenStaking.sol [hunk 10]
```diff
@@ -341,62 +350,73 @@ contract TokenStaking is Initializable, IStaking, Checkpoints {
...
 
     /// @notice Allows the Governance to set the minimum required stake amount.
     ///         This amount is required to protect against griefing the staking
     ///         contract and individual applications are allowed to require
     ///         higher minimum stakes if necessary.
-    /// @dev Operators are not required to maintain a minimum T stake all
+    /// @dev Providers are not required to maintain a minimum T stake all
     ///      the time. 24 hours after the delegation, T stake can be reduced
     ///      below the minimum stake. The minimum stake is just to protect
     ///      against griefing stake operation.
```
[the remark this answers is not in the document]
Comment (pdyraga) [reply contracts/staking/TokenStaking.sol:422]: Done in https://github.com/threshold-network/solidity-contracts/pull/71.

File: contracts/staking/TokenStaking.sol [hunk 11]
```diff
@@ -108,17 +108,17 @@ contract TokenStaking is Initializable, IStaking, Checkpoints {
...
-    mapping(address => OperatorInfo) internal operators;
+    mapping(address => StakingProviderInfo) internal stakingProviders;
     mapping(address => ApplicationInfo) public applicationInfo;
     address[] public applications;
 
     SlashingEvent[] public slashingQueue;
     uint256 public slashingQueueIndex;
 
-    event OperatorStaked(
+    event ProviderStaked(
```
[the remark this answers is not in the document]
Comment (pdyraga) [reply contracts/staking/TokenStaking.sol:118]: Or... just `Staked`.

[the remark this answers is not in the document]
Comment (pdyraga) [reply contracts/staking/TokenStaking.sol:118]: Done in #72

**Pull request merged**